linera-core/src/updater.rs

```diff
@@ -161,7 +159,7 @@ where
             && (highest_key_score >= committee.quorum_threshold()
                 || highest_key_score + remaining_votes < committee.quorum_threshold())
         {
-            end_time = Some(Instant::now() + start_time.elapsed().mul_f64(GRACE_PERIOD));
+            end_time = Some(Instant::now() + start_time.elapsed().mul_f64(grace_period));
```

Comment:
We could do this only in the first case (i.e. if we _did_ reach a quorum).

FYI I created https://github.com/linera-io/linera-protocol/issues/2836 for this.